src/field_theory/complete_ordered.lean

```diff
@@ -0,0 +1,605 @@
+          neg_neg], }, },
+    { simp only [mul_zero, add_monoid_hom.map_zero], },
+    { exact this x h, },
+  end,
+  intros x hpos,
+  -- prove that the (Sup of rationals less than x) ^ 2 is the Sup of the set of rationals less than
+  -- (x ^ 2) by showing it is an upper bound and any smaller number is not an upper bound
+  apply cSup_intro (cut_image_nonempty F K _),
+  { rintros a ha,
+    rw mem_cut_image_iff at ha,
+    rcases ha with ⟨q, rfl, ha⟩,
+    by_cases hq : 0 ≤ (q : K),
+    { have : 0 ≤ (q : F) := by exact_mod_cast hq,
+      obtain ⟨q2, hqpos, hq21, hq22⟩ := exists_rat_sqr_btwn ha this,
+      rw pow_two at hq22,
+      have : (q2 : F) < x := lt_of_mul_self_lt_mul_self (rat.cast_nonneg.mpr hqpos) hpos hq22,
+      rw ← @mem_cut_image_iff' F K at this,
+      have : (q2 : K) ≤ induced_map F K x := le_cSup _ _ (cut_image_bdd_above F K x) this,
+      transitivity (q2 : K)^2,
+      apply le_of_lt,
+      assumption_mod_cast,
+      rw pow_two,
+      have q2pos : (0 : K) ≤ q2 := by exact_mod_cast hqpos,
+      exact mul_le_mul this this q2pos (le_trans _ _ _ q2pos this), },
+    { transitivity (0 : K),
+      push_neg at hq,
+      exact le_of_lt hq,
+      exact mul_self_nonneg (Sup (cut_image F K x)), }, },
+  { intros y hy,
+    by_cases hypos : 0 ≤ y,
+    { obtain ⟨q2, hqpos, hq21, hq22⟩ := exists_rat_sqr_btwn hy hypos,
+      rw pow_two at hq22,
+      have : (q2 : K) < _ := lt_of_mul_self_lt_mul_self _ _ hq22,
+      use (q2 : K)^2,
+      split,
+      norm_cast,
+      rw mem_cut_image_iff',
+      erw [induced_add_map, lt_induced_map_iff] at this,
+      obtain ⟨q3, hq23, hh⟩ := this,
+      rw pow_two,
+      push_cast,
+      have : (q2 : F) < x :=
+      begin
+        transitivity (q3 : F),
+        assumption_mod_cast,
+      end,
+      apply mul_lt_mul'' this this,
+      assumption_mod_cast,
+      assumption_mod_cast,
+      exact hq21,
+      exact_mod_cast hqpos,
+      simp only [induced_add_map, add_monoid_hom.coe_mk],
+      rw lt_induced_map_iff,
+      obtain ⟨q3, q30, q3x⟩ := exists_rat_btwn hpos,
+      use q3,
+      split,
+      assumption_mod_cast, },
+    { use ((0 : ℚ) : K),
+      split,
+      rw [mem_cut_image_iff', rat.cast_zero],
+      exact linear_ordered_field.mul_pos _ _ hpos hpos,
+      push_neg at hypos,
+      rw [rat.cast_zero],
+      exact hypos, }, }
+end
+
+@[simp]
+lemma induced_map_inv_self (F K : Type*) [conditionally_complete_linear_ordered_field F]
+  [conditionally_complete_linear_ordered_field K] (x : F) :
+  induced_map K F (induced_map F K x) = x :=
+begin
+  rw [induced_map],
+  apply cSup_intro (cut_image_nonempty K F (induced_map F K x)),
+  { intros y hy,
+    rw mem_cut_image_iff at hy,
+    rcases hy with ⟨q, rfl, h⟩,
+    rw induced_map at h,
+    obtain ⟨y, hym, hyq⟩ := exists_lt_of_lt_cSup (cut_image_nonempty F K x) h,
+    rw mem_cut_image_iff at hym,
+    rcases hym with ⟨q2, rfl, h⟩,
```

Comment:
Think you could also use `obtain` here for consistency:
```suggestion
    obtain ⟨q2, rfl, h⟩ := hym,
```
not that it really matters